app/controllers/endpoints_controller.rb

```diff
@@ -37,11 +39,26 @@ def permitted_params
     )
   end
 
+  def update_params
+    params.permit(
+      :certificate_id,
+      :ssl_policy
+    )
+  end
+
   def load_endpoint
-    @endpoint = @district.endpoints.find_by!(name: params[:id])
+    @endpoint = @district.endpoints.find_by(name: params[:id])
+    unless @endpoint.present?
+      @endpoint = @district.endpoints.find_by!(name: name_prefix + params[:id])
+    end
```

Comment:
I think we will need another column for district. This is because its possible that there might exist a district called "food" and a endpoint called "bar". If there was a legacy endpoint called "food-bar", the endpoint names will collide and you will load the wrong endpoint!

To avoid this risk with other users of Barcelona (if any) lets add a column and it will be a lot better.

Reply:
You're absolutely right.
I agree with adding the new column.
let me add one

Reply:
@davidsiaw  I've been thinking about this issue.
district had already been added to the endpoint

https://github.com/degica/barcelona/blob/2734382417d3670bce5b43676f02c8eca27cb647/db/schema.rb#L53-L63

there is a unique validation for name so at least it should fail to create if there is a existing name which will collide

Reply:
The collision can cause other kinds of problems, not just on create, such as an endpoint pointing to the wrong servers. We should eliminate the issue before giving a chance for other issues to crop up.

Reply:
If the name is guaranteed to be unique when creating, I don't think it's possible to load the wrong endpoint.

https://github.com/degica/barcelona/blob/07009e6736178a39b36816517af55a766db585c8/app/models/endpoint.rb#L122-L126

When loading, it will look for an endpoint with a matching name in the specified district.
So I dont think it will load wrong one if endpoint is unique.

ok how about using `heritage.id` ?
I think this should decrease the collision with exiting one.

Reply:
@davidsiaw 
I added the tests for your concerns https://github.com/degica/barcelona/pull/660/commits/2c5cac8e7328a84e86a485d248cc1d51e6bd9df0

When a collision occurs, the creation of a new one is failing.
To reduce the possibility of a collision with an existing one or one with a district name, we can add an ID to the name, but I think it will be difficult to read the endpoint name when you check it in AWS.
Also, I wonder how much of that collision will happen.
Let me know what you think 🙏

Reply:
@Yuuki77 Ah, I don't mean the creation of an endpoint. We talked about that already, I am talking about

```
environments:
  staging:
    name: blue-green
    image_name: quay.io/dsiawdegica/bg
    scheduled_tasks: []
    services:
      - name: web
        service_type: web
        memory: 256
        command: sh /start.sh
        web_container_port: 8080
        listeners:
          - endpoint: staging-blue-green <-------------------------------------
            health_check_path: /

```

Reply:
@davidsiaw Thank you for the explanations
now it makes sense to me!
added the filter!
https://github.com/degica/barcelona/pull/660/commits/09477c57ea912d7157dfb6f9af8c4bb713240f23